tedana/model/fit.py

```diff
@@ -278,15 +269,133 @@ def fitmodels_direct(catd, mmix, mask, t2s, t2s_full, tes, combmode, ref_img,
                 threshold=(max(tsoc_Babs.shape) - countsigFS0), mask=mask,
                 binarize=True)
 
+        if method == 'kundu_v2':
+            # WTS, tsoc_B, PSC, and F_S0_maps are not used by Kundu v2.5
+            selvars = ['Z_maps', 'F_R2_maps',
+                       'Z_clmaps', 'F_R2_clmaps', 'F_S0_clmaps',
+                       'Br_R2_clmaps', 'Br_S0_clmaps']
+        elif method == 'kundu_v3':
+            selvars = ['WTS', 'tsoc_B', 'PSC',
+                       'Z_maps', 'F_R2_maps', 'F_S0_maps',
+                       'Z_clmaps', 'F_R2_clmaps', 'F_S0_clmaps',
+                       'Br_R2_clmaps', 'Br_S0_clmaps']
+        elif method is None:
+            selvars = []
+        else:
+            raise ValueError('Method "{0}" not recognized.'.format(method))
+
         seldict = {}
-        selvars = ['WTS', 'tsoc_B', 'PSC',
-                   'Z_maps', 'F_R2_maps', 'F_S0_maps',
-                   'Z_clmaps', 'F_R2_clmaps', 'F_S0_clmaps',
-                   'Br_R2_clmaps', 'Br_S0_clmaps']
         for vv in selvars:
             seldict[vv] = eval(vv)
+    else:
+        seldict = None
+
+    return comptable, seldict, betas, mmix_new
+
+
+def kundu_metrics(comptable, metric_maps):
```

Comment:
If there's an eventual plan to put it in there, it might not hurt to stub it and place a %TODO under it, at least in part so there's a prototype for how you would slot in a new method after this refactor.